compiler/rustc_traits/src/chalk/lowering.rs

```diff
@@ -370,7 +370,7 @@ impl<'tcx> LowerInto<'tcx, chalk_ir::Ty<RustInterner<'tcx>>> for Ty<'tcx> {
             ty::Placeholder(_placeholder) => {
                 chalk_ir::TyKind::Placeholder(chalk_ir::PlaceholderIndex {
                     ui: chalk_ir::UniverseIndex { counter: _placeholder.universe.as_usize() },
-                    idx: _placeholder.name.as_usize(),
+                    idx: _placeholder.name.expect_anon() as usize,
```

Comment:
I'm pretty sure that due to canonicalization, this will never be triggered. 

In any case, I'm happpy with not having chalk support for bound types :grimacing: -- we also treat RePlaceholder very strangely below (always giving it an `idx: 0`)